nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/convolution/DepthwiseConv2DBp.java

```diff
@@ -0,0 +1,241 @@
+        Map<String, Map<String, PropertyMapping>> ret = new HashMap<>();
+        Map<String, PropertyMapping> map = new HashMap<>();
+        val strideMapping = PropertyMapping.builder()
+                .tfAttrName("strides")
+                .onnxAttrName("strides")
+                .propertyNames(new String[]{"sW", "sH"})
+                .build();
+
+
+        val kernelMappingH = PropertyMapping.builder()
+                .propertyNames(new String[]{"kH"})
+                .tfInputPosition(1)
+                .shapePosition(0)
+                .onnxAttrName("kernel_shape")
+                .build();
+
+        val kernelMappingW = PropertyMapping.builder()
+                .propertyNames(new String[]{"kW"})
+                .tfInputPosition(1)
+                .shapePosition(1)
+                .onnxAttrName("kernel_shape")
+                .build();
+
+        val dilationMapping = PropertyMapping.builder()
+                .onnxAttrName("dilations")
+                .propertyNames(new String[]{"dW", "dH"})
+                .tfAttrName("rates")
+                .build();
+
+        val dataFormat = PropertyMapping.builder()
+                .onnxAttrName("data_format")
+                .tfAttrName("data_format")
+                .propertyNames(new String[]{"dataFormat"})
+                .build();
+
+        val nhwc = PropertyMapping.builder()
+                .onnxAttrName("data_format")
+                .tfAttrName("data_format")
+                .propertyNames(new String[]{"isNHWC"})
+                .build();
+
+        val sameMode = PropertyMapping.builder()
+                .onnxAttrName("auto_pad")
+                .propertyNames(new String[]{"isSameMode"})
+                .tfAttrName("padding")
+                .build();
+
+        val paddingWidthHeight = PropertyMapping.builder()
+                .onnxAttrName("padding")
+                .propertyNames(new String[]{"pH", "pW"})
+                .build();
+
+
+        map.put("sW", strideMapping);
+        map.put("sH", strideMapping);
+        map.put("kH", kernelMappingH);
+        map.put("kW", kernelMappingW);
+        map.put("dW", dilationMapping);
+        map.put("dH", dilationMapping);
+        map.put("isSameMode", sameMode);
+        map.put("pH", paddingWidthHeight);
+        map.put("pW", paddingWidthHeight);
+        map.put("dataFormat", dataFormat);
+
+
+        return ret;
+    }
+
+
+    @Override
+    public String opName() {
+        return "depthwise_conv2d_bp]";
+    }
+
+
+    @Override
+    public List<DataType> calculateOutputDataTypes(List<DataType> inputDataTypes) {
+        int n = args().length;
+        Preconditions.checkState(inputDataTypes != null && inputDataTypes.size() == n, "Expected %s input data types for %s, got %s", n, getClass(), inputDataTypes);
+        return Collections.singletonList(inputDataTypes.get(0));
+    }
+
+    @Override
+    public int getNumOutputs(){return 1;}
```

Comment:
Incorrect. Does not match c++ op.
https://github.com/eclipse/deeplearning4j/blob/master/libnd4j/include/ops/declarable/generic/nn/convo/depthwiseConv2d.cpp#L157-L159
op has either 2 or 3 outputs, depending on 
we can do something like `return args().length == 4 ? 3 : 2`
as args().length == 4 when bias is provided, or length 3 otherwise (input/weights/bias/gradO vs. input/weights/gradO)